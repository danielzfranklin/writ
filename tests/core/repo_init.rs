use test_support::*;

#[test]
fn can_init() -> Result {
    init();
    let (dir, _repo) = repo_fixture()?;
    assert_debug_snapshot!(all_entries(dir.path().join(".git"))?);

    Ok(())
}

#[test]
fn can_init_nonexistent_ws() -> Result {
    init();
    let dir = tempdir()?;
    let subdir = dir.path().join("subdir");

    let _repo = Repo::init(&subdir)?;
    assert_debug_snapshot!(all_entries(subdir.join(".git"))?);

    Ok(())
}
