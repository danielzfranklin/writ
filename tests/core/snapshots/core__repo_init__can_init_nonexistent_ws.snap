---
source: tests/core/repo_init.rs
expression: "all_entries(subdir.join(\".git\"))?"

---
[
    "objects",
    "refs",
]
